File: src/main/java/com/github/holly/accountability/validation/BindingResultWrapper.java
package com.github.holly.accountability.validation;
import org.springframework.validation.BindingResult;

import java.util.Arrays;
import java.util.List;


public class BindingResultWrapper {
    private List<BindingResultError> violations;

    public BindingResultWrapper(BindingResult bindingResult) {
       this.violations = bindingResult.getFieldErrors().stream().map( res -> {
           if(res.getCodes() == null){
             return new BindingResultError(res.getField(), List.of());
           }
           return new BindingResultError(res.getField(), Arrays.asList(res.getCodes()));
       }).toList();
    }

    public List<BindingResultError> getViolations() {
        return violations;
    }
}

File: src/main/java/com/github/holly/accountability/validation/BindingResultWrapper.kt


// TODO: convert all kotlin files to java file and code! (for uniformity)

package com.github.holly.accountability.validation

import org.springframework.validation.BindingResult

class BindingResultWrapper(bindingResult: BindingResult) {
    val violations: List<BindingResultError> = bindingResult.fieldErrors.map { BindingResultError(it.field, it.codes?.toList() ?: listOf()) }
}